src/Adapter/State/QueryHandler/IsUniqueStateIsoCodeHandler.php

```diff
@@ -0,0 +1,57 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\State\QueryHandler;
+
+use PrestaShop\PrestaShop\Core\Domain\State\Query\isUniqueStateIsoCode;
+use PrestaShop\PrestaShop\Core\Domain\State\QueryHandler\IsUniqueStateIsoCodeHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\State\QueryResult\IsFoundStateByIsoCode;
+use State;
+
+/**
+ * Handles query for determining if state with given iso code exists
+ */
+final class IsUniqueStateIsoCodeHandler implements IsUniqueStateIsoCodeHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(isUniqueStateIsoCode $query): IsFoundStateByIsoCode
+    {
+        /** @var int $state */
+        $state = State::getIdByIso($query->getIsoCode());
+
+        if (!$state) {
+            return new IsFoundStateByIsoCode(false);
+        }
+
+        if ($query->getStateId() !== null && $state != $query->getStateId()) {
```

Comment:
Strict comparison in this place will always fail because State::getIdByIso($query->getIsoCode()) returns statet id as string and $query->getStateId() returns int. And casting string to int just because strict comparison could be used seemed useless